pkg/pillar/hypervisor/kvm.go

```diff
@@ -940,6 +951,20 @@ func (ctx KvmContext) CreateDomConfig(domainName string,
 		}
 	}
 
+	// render vsock settings, this should go last to avoid
+	// PCI ID conflicts, let qemu assign PCI ID for vsock.
+	//
+	// clientCid needs atomic add to avoid race condition
+	// in case CreateDomConfig is called concurrently, which
+	// happens at least in unit tests.
+	vsockContext := struct {
+		GuestCID string
+	}{GuestCID: fmt.Sprintf("%d", atomic.AddUint32(&clientCid, 1))}
```

Comment:
Don't we want to store the value in DomainConfig or DomainStatus? With the current implementation, the channel ID is lost once created, and we can never identify a channel for a given VM. However, it can be useful in the future if we add some interaction initialized by the host. Or, if we do not expect any interaction initialized by the host, it would be nice to note it explicitly in the documentation.

Reply:
I don't think we expect the host to connect to the client, then we have to impose on every app to have a vcom client running.

Reply:
Could you please add it as a comment? A short clarification on why "fire and forget" is fine here.

Reply:
Up! Still wanna see it in the comment...

Reply:
done.